### PR TITLE
:computer: Fixes #59 Map internal forces to nodes

### LINKED ISSUE #59: Compute internal forces at nodes

### PR DESCRIPTION
**Pull request by kks32**
* Convert particle stresses to internal forces at nodes

### CODE REVIEW

File: include/cell.tcc [hunk 1]
```diff
@@ -501,37 +502,33 @@ template <unsigned Tdim>
...
   Eigen::VectorXd shapefns = shapefn_->shapefn(xi);
+  // Map external forces from particle to nodes
   for (unsigned i = 0; i < this->nfunctions(); ++i)
     nodes_[i]->update_external_force(true, nphase,
                                      shapefns(i) * pgravity * pmass);
 }
 
-/*
-//! Assign internal force to nodes
+//! Map internal force to nodes in a 2D cell
```
Comment (cbgeo) [inline include/cell.tcc]: Both for 2D and 3D cells

File: include/cell.tcc [hunk 2]
```diff
@@ -501,37 +502,33 @@ template <unsigned Tdim>
...
-         temp_stress(2) = pstress(3,i);
-      }
-      else if (Tdim == 3)
-         temp_stress = pstress.col(i);
-      for (unsigned j = 0; j < this->nfunctions(); ++j)
-          nodes_[j]->update_internal_force(i, (pvolume *
-B_matrix.at(j).transpose() * temp_stress));
+void mpm::Cell<Tdim>::map_internal_force_to_nodes(
+    unsigned nphase, double pvolume, const VectorDim& xi,
+    const Eigen::VectorXd& pstress) {
```
Comment (cbgeo) [inline include/cell.tcc]: Check if `pstress` is the right size for the dimension?

File: tests/cell_test.cc [hunk 3]
```diff
@@ -185,17 +186,52 @@ TEST_CASE("Cell is checked for 2D case", "[cell][2D]") {
           REQUIRE(node->momentum(phase)(i) == Approx(1.0).epsilon(Tolerance));
       }
     }
+
     SECTION("Check particle body force mapping") {
       // Assign body force to nodes
       cell->map_body_force_to_nodes(xi, phase, pmass, pgravity);
-      Eigen::Vector2d bodyforce;
+      Eigen::Vector3d bodyforce;
```
Comment (cbgeo) [inline tests/cell_test.cc]: Shouldn't this be just 2D?

File: tests/particle_test.cc [hunk 4]
```diff
@@ -102,8 +102,7 @@ TEST_CASE("Particle is checked for 1D case", "[particle][1D]") {
     REQUIRE(particle->mass(Phase) == Approx(100.5).epsilon(Tolerance));
 
     // Check stress
-    Eigen::VectorXd stress;
-    stress.resize(Dim);
+    Eigen::Matrix<double, 6, 1> stress;
```
Comment (cbgeo) [inline tests/particle_test.cc]: I guess we will never have a 1D test for MPM with particles!

**Review by cbgeo — CHANGES_REQUESTED** [review]

File: tests/particle_test.cc (hunk 4 above)
Comment (kks32) [reply tests/particle_test.cc]: Easier fix than to remove all tests for 1D particle

Comment (kks32) [reply tests/particle_test.cc]: Maybe removal should be handled in a separate PR?

**Review by cbgeo — APPROVED** [review]

**Pull request merged**